spring-batch-core/src/test/java/org/springframework/batch/core/step/tasklet/MethodInvokingTaskletAdapterTest.java

```diff
@@ -0,0 +1,239 @@
+			this.chunkContext = chunkContext;
+			return RepeatStatus.FINISHED;
+		}
+		
+		/* different parameters order */
+		public RepeatStatus execute2(ChunkContext chunkContext, StepContribution stepContribution) throws Exception {
+			this.stepContribution = stepContribution;
+			this.chunkContext = chunkContext;
+			return RepeatStatus.FINISHED;
+		}
+		
+		/* subset of arguments: only chunk context */
+		public RepeatStatus execute3(ChunkContext chunkContext) throws Exception {
+			this.chunkContext = chunkContext;
+			return RepeatStatus.FINISHED;
+		}
+		
+		/* subset of arguments: only step contribution */
+		public RepeatStatus execute4(StepContribution stepContribution) throws Exception {
+			this.stepContribution = stepContribution;
+			return RepeatStatus.FINISHED;
+		}
+		
+		/* subset of arguments: no arguments */
+		public RepeatStatus execute5() throws Exception {
+			return RepeatStatus.FINISHED;
+		}
+		
+		/* compatible return type: boolean */
+		public boolean execute6(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+			this.stepContribution = stepContribution;
+			this.chunkContext = chunkContext;
+			return true;
+		}
+		
+		/* compatible return type: void */
+		public void execute7(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+			this.stepContribution = stepContribution;
+			this.chunkContext = chunkContext;
+		}
+		
+		/* subset of arguments (only step contribution) and compatible return type (boolean) */
+		public boolean execute8(StepContribution stepContribution) throws Exception {
+			this.stepContribution = stepContribution;
+			return true;
+		}
+		
+		/* subset of arguments (only chunk context) and compatible return type (void) */
+		public void execute9(ChunkContext chunkContext) throws Exception {
+			this.chunkContext = chunkContext;
+		}
+		
+		/* Incorrect signature: extra parameter (ie a superset not a subset as specified) */
+		public RepeatStatus execute10(StepContribution stepContribution, ChunkContext chunkContext, String string) throws Exception {
+			this.stepContribution = stepContribution;
+			this.chunkContext = chunkContext;
+			return RepeatStatus.FINISHED;
+		}
+		
+		/* ExitStatus return type : should be returned as is */
+		public ExitStatus execute11(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+			this.stepContribution = stepContribution;
+			this.chunkContext = chunkContext;
+			return new ExitStatus("DONE");
+		}
+		
+		/* Non ExitStatus return type : should return ExitStatus.COMPLETED */
+		public String execute12(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+			this.stepContribution = stepContribution;
+			this.chunkContext = chunkContext;
+			return "DONE";
+		}
+		
+		public StepContribution getStepContribution() {
+			return stepContribution;
+		}
+		
+		public ChunkContext getChunkContext() {
+			return chunkContext;
+		}
+	}
+}
```

Comment:
Please add a new line at the end of the file

Reply:
Done.